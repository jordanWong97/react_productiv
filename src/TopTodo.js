import React from "react";

import Todo from "./Todo";

/** Shows the top todo.
 *
 * Props:
 * - todos
 *
 * TodoApp -> TopTodo
 */

function TopTodo({ todos }) {
  // lowest-priority # is the highest priority
  let top = todos.reduce(
    (acc, cur) => cur.priority < acc.priority ? cur : acc, todos[0]);

  const { id, title, description, priority } = top;

  return (
    <div>
      {top &&
        <Todo id={id}
          title={title}
          description={description}
          priority={priority} />
      }
    </div>
  );
}

export default TopTodo;