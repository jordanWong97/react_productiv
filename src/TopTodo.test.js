import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";

describe("productiv app", function () {
  it("renders without crashing", function () {
    render(<TopTodo todos={[{
      id: 1,
      title: "test",
      description: "test",
      priority: 1
    }]} />);
  });

  it("renders todo title", function () {
    const { container, queryByText } = render(<TopTodo todos={[{
      id: 1,
      title: "testTitle",
      description: "testDesc",
      priority: 1
    }]} />);
    expect(queryByText("testTitle")).toBeInTheDocument();
  });

  it("renders todo description", function () {
    const { container, debug, queryByText } = render(<TopTodo todos={[{
      id: 1,
      title: "testTitle",
      description: "testDesc",
      priority: 1
    }]} />);
    expect(queryByText("testDesc")).toBeInTheDocument();
  });

  it("snapshot test", function () {
    const { container } = render(<TopTodo todos={[{
      id: 1,
      title: "testTitle",
      description: "testDesc",
      priority: 1
    }]} />);
    expect(container).toMatchSnapshot();
  });

});

// todo={{
//   id: 1,
//   title: "test",
//   description: "test",
//   priority: 1
// }}

