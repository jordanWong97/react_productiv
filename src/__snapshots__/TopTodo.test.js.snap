// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv app snapshot test 1`] = `
<div>
  <div>
    <div
      class="Todo"
    >
      <div>
        <b>
          testTitle
        </b>
         
        <small>
          (priority: 
          1
          )
        </small>
      </div>
      <div>
        <small>
          testDesc
        </small>
      </div>
    </div>
  </div>
</div>
`;
